mono/profiler/proflog.c

```diff
@@ -1619,6 +1622,74 @@ gc_handle (MonoProfiler *prof, int op, int type, uintptr_t handle, MonoObject *o
 	process_requests (prof);
 }
 
+static void
+finalize_begin (MonoProfiler *prof)
+{
+	ENTER_LOG;
+
+	LogBuffer *buf = ensure_logbuf (
+		EVENT_SIZE /* event */
+	);
+
+	emit_event (buf, TYPE_GC_FINALIZE_START | TYPE_GC);
+
+	EXIT_LOG;
+
+	process_requests (prof);
```

Comment:
Maybe we should only trigger a heapshot at a deterministic point of finalization? Either at begin or end.

IDK whether triggering at any point would not produce some weird artifacts.